testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java

```diff
@@ -6422,6 +6422,10 @@ private static void checkIf(SqlOperatorFixture f) {
       f.checkScalarApprox("atanh(0.76159416)", "DOUBLE NOT NULL",
           isWithin(1d, 0.0001d));
       f.checkScalarApprox("atanh(cast(-0.1 as decimal))", "DOUBLE NOT NULL",
+          isWithin(-0.0d, 0.0001d));
```

Comment:
The SqlOperatorTest has a lot of test cases with casts to decimals etc that are currently disabled due to `SqlOperatorTest#DECIMAL` field. Are we planning to fix those as part of this PR? It looks like we should.

Note that currently there is a bug (tracked under CALCITE-5923) and due to that `CalciteSqlOperatorTest` are not running as expected.

Reply:
@pfzhan This question is still open. How we should treat the test cases guarded by SqlOperatorTest#DECIMAL?

Reply:
If we enable the tests under `SqlOperatorTest#DECIMAL` do we need the new tests that were added in this class by this PR? Do they offer additional test coverage?

Reply:
I run the test found that SqlOperatorTest does not compare with expected result, but the CalciteSqlOperatorTest class does. Some cases were add to explain slight changes, the test in .iq was implicit, but this is explicit.

I checked the pr of [CALCITE-5923](https://issues.apache.org/jira/browse/CALCITE-5923). It seems that handles date/timestamp cases rather than decimal type.

Reply:
The `SqlOperatorTest` class contains many tests casting various types to decimals. These tests are all disabled since `SqlOperatorTest#DECIMAL` is `false`.

Since in this PR you are basically fixing CAST to DECIMAL type, I was thinking that it makes sense to enable those that are affected by this fix. If the existing tests are sufficient to test the changes in this PR then it is not really necessary to add new ones (using `atanh` etc.).

Basically, another way to put my question is the following.
* Can we set SqlOperatorTest#DECIMAL to true ?
* Can we remove some `if (DECIMAL)`, `if (!DECIMAL)` statements?
* Does this PR fix any of the above?

Reply:
Got it, this will be considered in this pr.